apex/transformer/functional/fused_softmax.py

```diff
@@ -162,7 +162,10 @@ def is_kernel_available(self, mask, b, np, sq, sk):
         if (
             self.scaled_masked_softmax_fusion  # user want to fuse
             and self.input_in_float16  # input must be fp16
-            and mask is not None  # mask tensor must not be None
+            and (
+                self.attn_mask_type == AttnMaskType.causal
+                or (self.attn_mask_type == AttnMaskType.padding and mask is not None)
+            )
```

Comment:
does this check happen to be what @yidong72 suggested?

Reply:
@yidong72 has a PR open to NeMo for his changes: https://github.com/NVIDIA/NeMo/pull/4594

Reply:
then what is this check for? clarity?